error_logger

mitchellhenke opened the pull request.

DavidAntaramian commented on lib/timber/events/error_event.ex:
```diff
@@ -137,6 +155,26 @@ defmodule Timber.Events.ErrorEvent do
+      end
+
+      file = Keyword.get(location, :file)
+             |> Kernel.to_string()
```
Just for clarity, is this because the result of `Keyword.get(location, :file)` would be an IO list?

mitchellhenke replied:
yep

DavidAntaramian commented on lib/timber/integrations/error_logger.ex:
```diff
@@ -0,0 +1,352 @@
+
+  However, the OTP report handler handles additional report types as well. If you
+  find that you would like these reports to be logged, just be aware that every
+  exception will be displayed twice in the log.
```
I think you should add a note here saying that, with the default handler the report will also be broken out across multiple lines; whereas the Timber handler consolidates the information into a single line

DavidAntaramian commented on lib/timber/integrations/error_logger.ex:
```diff
@@ -0,0 +1,352 @@
+  def init(opts \\ []) do
+    max_message_count = Keyword.get(opts, :max_message_count, 300)
+    state = %{
+      logger: Process.whereis(Logger),
```
For the sake of future maintainers, could you add a note before all the function declarations about what each state field is used for and how the overall architecture of this works?

DavidAntaramian commented on lib/timber/integrations/error_logger.ex:
```diff
@@ -0,0 +1,352 @@
+    {:ok, state}
+  end
+
+  def get_metadata({:crash_report, [error_report | _neighbors]}, :error) do
```
I thought we were ignoring crash reports altogether, so I'm confused as to what this code block is responsible for.

DavidAntaramian commented:
Oh, and please add an entry to CHANGELOG.md

mitchellhenke commented:
@DavidAntaramian Addressed your comments, let me know if it isn't clear enough.

Wasn't sure if I should squash the commit before you get a chance to review.

binarylogic commented:
Yes, please squash.

mitchellhenke commented:
done!

binarylogic commented:
I'll let @DavidAntaramian approve and merge
